packages/mds-db/index.ts

```diff
@@ -383,28 +383,20 @@ async function updateDevice(device_id: UUID, changes: Partial<Device>): Promise<
   })
 }
 
-async function writeEvent(event_param: VehicleEvent): Promise<Recorded<VehicleEvent>> {
+async function writeEvent(event_param: VehicleEvent) {
   const device = await readDevice(event_param.device_id)
-  return new Promise((resolve, reject) => {
-    if (!device) {
-      reject(new Error('device unregistered'))
-    } else {
-      // write pg
-      getWriteableClient().then(client => {
-        const telemetry_timestamp = event_param.telemetry ? event_param.telemetry.timestamp : null
-        const event = { ...event_param, telemetry_timestamp }
-        const sql = `INSERT INTO ${cols_sql(schema.EVENTS_TABLE, schema.EVENTS_COLS)} ${vals_sql(schema.EVENTS_COLS)}`
-        const values = vals_list(schema.EVENTS_COLS, event)
-        logSql(sql, values)
-        client
-          .query(sql, values)
-          .then(() => {
-            resolve(event as Recorded<VehicleEvent>)
-          }, reject)
-          .catch(reject)
-      })
-    }
-  })
+  if (!device) {
```

Comment:
I am pretty sure `readDevice` will `reject` so this `!device` may never happen. You should test that.

Reply:
You're correct! I think this is an artifact from long ago.